syntax = "proto3";

package pero;

import "google/protobuf/timestamp.proto";

message Data {
    string name = 1;
    bytes content = 2;  // todo - enum for url
}

message StageLog {
    string host_id = 1;
    string stage = 2;
    google.protobuf.Timestamp start = 3;
    google.protobuf.Timestamp end = 4;
    string status = 5;  // overall status of processing
    string log = 6;
}

message ProcessingRequest {
    string uuid = 1;
    string page_uuid = 2; // identifies page / image
    int32 priority = 3;
    google.protobuf.Timestamp start_time = 4;
    repeated string processing_stages = 5;
    repeated Data results = 6;  // image is in data - 1
    repeated StageLog logs = 7;
}
